rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/60_dense_vector_dynamic_mapping.yml

```diff
@@ -0,0 +1,339 @@
+          ]
+
+  - do:
+      indices.get_mapping:
+        index: my-dynamic-template-index
+
+  - match: { my-dynamic-template-index.mappings.properties.my_dense_vector_field.type: dense_vector }
+  - match: { my-dynamic-template-index.mappings.properties.my_dense_vector_field.dims: 5 }
+  - match: { my-dynamic-template-index.mappings.properties.my_dense_vector_field.index: true }
+  - match: { my-dynamic-template-index.mappings.properties.my_dense_vector_field.similarity: cosine }
+  - match: { my-dynamic-template-index.mappings.properties.my_small_float_field.type: float }
+  - match: { my-dynamic-template-index.mappings.properties.my_float_field.type: float }
+  - match: { my-dynamic-template-index.mappings.properties.my_dynamic_field.type: dense_vector }
+  - match: { my-dynamic-template-index.mappings.properties.my_dynamic_field.dims: 128 }
+  - match: { my-dynamic-template-index.mappings.properties.my_dynamic_field.index: true }
+  - match: { my-dynamic-template-index.mappings.properties.my_dynamic_field.similarity: cosine }
+
+---
+"Fields mapped as dense_vector without dims dynamically map the dim size to the first indexed document":
+
+  - do:
+      indices.create:
+        index: test-mapped-index
+        body:
+          settings:
+            number_of_shards: 1
+            number_of_replicas: 0
+          mappings:
+            properties:
+              my_dense_vector_field:
+                type: dense_vector
+
+  - do:
+      indices.get_mapping:
+        index: test-mapped-index
+
+  - match: { test-mapped-index.mappings.properties.my_dense_vector_field.type: dense_vector }
+  - match: { test-mapped-index.mappings.properties.my_dense_vector_field.index: true }
+  - match: { test-mapped-index.mappings.properties.my_dense_vector_field.similarity: cosine }
+  - match: { test-mapped-index.mappings.properties.my_dense_vector_field.dims: null }
+
+  - do:
+      index:
+        index: test-mapped-index
+        id: "1"
+        refresh: true
+        body:
+          my_dense_vector_field: [ 159.1, 289.56, -128.7424 ]
+
+  - do:
+      indices.get_mapping:
+        index: test-mapped-index
+
+  - match: { test-mapped-index.mappings.properties.my_dense_vector_field.type: dense_vector }
+  - match: { test-mapped-index.mappings.properties.my_dense_vector_field.index: true }
+  - match: { test-mapped-index.mappings.properties.my_dense_vector_field.similarity: cosine }
+  - match: { test-mapped-index.mappings.properties.my_dense_vector_field.dims: 3 }
```

Comment:
So, I added the following test locally, and it fails:
```
---
"Fields with sub-object float arrays within the threshold map as dense_vector":

  - do:
      index:
        index: test-map-dense-vector-deal
        id: "1"
        refresh: true
        body:
          parent_field:
            my_field: [
              159.1, 289.56, -128.7424, 145.9871, -164.0003, 86.4034, -89.6929, 257.9717, 131.6075, 67.9233, -144.8255, 223.8446, 77.3228, -210.1163, -139.4783, 12.6499, 15.4491, 108.3465, -189.3947, 178.2045, -187.5925, 184.5089, 77.3022, -202.7439, -13.4959, 115.9719, -139.4332, 196.7845, 104.7573, -156.7746, 166.9878, 68.3936, 159.8473, -141.4446, 21.1947, 186.5908, -209.6895, 68.6169, 44.1255, 147.4659, 56.5079, -179.7997, -85.1651, 11.4847, 124.1662, 96.2246, -178.6705, 85.5925, 205.3616, -16.4704, 172.4947, -115.2535, -58.1722, 94.4836, 34.6458, -70.1011, -58.8047, 149.9562, -37.8998, 196.9805, -169.3555, -163.9432, 188.5611, 214.8378, 29.3182, -24.8724, 152.9382, -109.4345, -123.6716, -8.2441, 64.5902, 27.8083, 40.8185, -94.3161, 58.1463, -138.7432, 24.6805, -88.7222, -11.2018, 206.6434, 201.9024, 87.3079, -3.2883, -60.2484, -109.5789, 105.5766, -116.6709, -17.7073, -71.5093, -75.2937, -176.8691, -146.4967, 53.7586, 199.5294, 55.9754, -48.7399, 82.2051, 135.2921, 22.4408, -116.4008, -33.7538, 29.7207, 6.3692, -97.5768, -12.7982, -200.9331, -62.2743, 81.0843, 136.2247, 150.2565, 139.6838, 155.2657, -25.7447, 198.5955, 18.8099, 46.9014, -60.2672, 136.4801, 171.8966, 172.5842, 13.9123, 75.8386, -64.2444, -48.1964, 135.9685, 7.4927, -40.6424, -76.8922
            ]


  - do:
      indices.get_mapping:
        index: test-map-dense-vector-deal

  - match: { test-map-dense-vector.mappings.properties.parent_field.properties.my_field.type: dense_vector }
  - match: { test-map-dense-vector.mappings.properties.parent_field.properties.my_field.dims: 128 }
  - match: { test-map-dense-vector.mappings.properties.parent_field.properties.my_field.index: true }
  - match: { test-map-dense-vector.mappings.properties.parent_field.properties.my_field.similarity: cosine }

```

I am digging into the cause currently. I am not sure, but it may have to do with mapper names and merges.

Reply:
Arg, I see the cause. 

everything in java is a reference. 

```
    public void setDynamicMappers(List<Mapper> dynamicMappers) {
        this.dynamicMappers = dynamicMappers;
    }
```

And the context is potentially wrapped. So, instead of updating the originally passed arraylist, this is actually changing the reference. Which would break if we exited the current parsing context wrapper.

:/

Reply:
Argh, I hadn't thought of that case, thank you